### PR TITLE
Add Support for front matter tag author: to add a by line to posts, w…

### PR DESCRIPTION
**Pull request by sschwetz**
(Resubmitted on correct source branch)

This pull request adds the ability to tag a post with an author's name by using the front matter author: field.

this pull adds/changes 5 files:

_config.yml - adds collection authors; adds scope for _author layout
_authors/stephen.md - example author's file
includes/header.html - adds logic to add link to posts
_layouts/author.html - layout used to generate authors bio pages
_layouts/home.html - adds by line to posts on the home page
_layouts/author.html needs some love to make it look pretty, UIX is not my forte.

### CODE REVIEW

File: _layouts/home.html [hunk 1]
```diff
@@ -45,10 +45,15 @@ <h3 class="post-subtitle">
           </h3>
         {% endif %}
       </a>
-
+
```
Comment (daattali) [inline _layouts/home.html:48]: Trim the whitespace

**Comment by daattali** [comment]
Since this PR introduces additional pages that are getting linked, it will require a bit more testing that most simpler features. Have you tested and ensured the links work on: GitHub user page, GitHub project page, local installations?

**Comment by daattali** [comment]
I like the idea of this feature, but I think it needs some work to get it to be production ready.

- I just tested this locally and the links do not work correctly. It tried to go to `http://authors/dean.html`
- Ignore the author on a page if that person doesn't have a matching page in the collection
- If a page has an autho, add the `author` meta tag 
- I looked at many websites and noticed that the standard is the place "By: author" on a separate line, before the date. So let's put it in a line before the date.
- For cleaner UI, the author name colour should match the colour of the rest of the metadata - let's make it **bold** to signify that it's clickable
- Documentation needs to be added

**Comment by sschwetz** [comment]
I have started adding the additions/alterations you asked for. From my googling, it doesn't look like Liquid has a way of checking to see if a file exists for the authors. 

As page.author is the frontmatter we are using, I would suspect that an end user would expect to see their name come up even if they didn't set up the collection for authors. Instead of not putting the byline there, how about we put the author's name in, but if their name does not exist in the collection, we don't create the name as a link?

**Comment by daattali** [comment]
To check if the author exists, I think this should work:

```
{% assign author_ids = site.authors | map: "username" %}
{% if post.author and author_ids contains post.author %} ...
```

Although now that you brought up the idea of showing the author even if no page exists - I'm thinking that maybe that should always be the case, for simplicity sake. Get rid of the collection, get rid of additional files per author. Simply allow you to add a `author: name` to a post, and if that field exists, then show the authors name. No other info about the author is required. You could create a `authors.md` page that loops through all the posts and lists all the authors, similar to the `tags` page. I know it's not as feature-rich as your original idea, but it does simplify the code a lot which is a major factor, and it still achieves the main objective of allowing names. What do you think?

**Comment by sschwetz** [comment]
Yeah that seems like a much better idea. I will set it up so that it will look for author: or authors: just to cover all bases in case a user thinks that they should use the plural if they have multiple authors

**Comment by daattali** [comment]
Let's have only `author`, and [in the documentation ](https://github.com/daattali/beautiful-jekyll#supported-parameters) it can be highlighted that the field can be a single value or a list

**Comment by daattali** [comment]
This PR seems to have more files than it needs to :)

**Comment by sschwetz** [comment]
Hmm, I don't know how I managed to do that. Reverted head and forced push seem to have fixed it. I have also merged the changes in the master branch into this commit to resolve the conflict with the font-awesome change

**Comment by daattali** [comment]
I've made some more changes and merging now, thanks

**Pull request merged**